drivers/timer/ite_it8xxx2_timer.c

```diff
@@ -21,6 +22,7 @@ LOG_MODULE_REGISTER(timer, LOG_LEVEL_ERR);
 #define EVENT_TIMER_FLAG	DT_INST_IRQ_BY_IDX(0, 0, flags)
 /* Event timer max count is 512 sec (base on clock source 32768Hz) */
 #define EVENT_TIMER_MAX_CNT	0x00FFFFFFUL
+#define EVENT_TIMER_FOREVER	(K_TICKS_FOREVER & BIT_MASK(31))
```

Comment:
Threre is an easy way to check the parameter "int32_t ticks" of sys_clock_set_timeout(), you can run the tests/kernel/timer/timer_api test on it8xxx2_evb, and print "int32_t ticks", then it shows 0x7fffffff when k_timer_start() is called.

Reply:
ps. It always inputs K_FOREVER period to k_timer_start()

Reply:
`CONFIG_TIMEOUT_64BIT = y`, 
then `k_ticks_t type is int64_t`. 
Magic macro transits `K_FOREVER` to `(k_timeout_t) { .ticks = (K_TICKS_FOREVER) }`, 
and there is a `#define K_TICKS_FOREVER ((k_ticks_t) -1)`, 
so `K_FOREVER` is a k_timeout_t type structure, the member `ticks` type is int64_t and value is 0xFFFF FFFF FFFF FFFF.

Reply:
I add these info to commit message.

Reply:
So in this configuration, if I am reading this correctly

> K_TICKS_FOREVER == 0xFFFF FFFF FFFF FFFF
> K_FOREVER == 0xFFFF FFFF FFFF FFFF

Which is like the original int64_t example above and when this is passed to a function that casts that value to an int32_t or just a straight cast, my gcc converted this to 0xFFFF FFFF instead of the expected value that you are checking for of 0x7FFF FFFF

Reply:
Yes, you're reading correctly. 

BTW, according to sys_clock_set_timeout() API definition: 
A ticks value of zero (or even negative) is legal and treated identically: it simply indicates the kernel would like the next tick announcement as soon as possible.
So passing ticks == 0xFFFF FFFF is equal passing ticks == 1.

Reply:
Ack